pxtwinrt/serial.ts

```diff
@@ -96,23 +121,40 @@ namespace pxt.winrt {
             return;
         }
 
+        const streams = Windows.Storage.Streams;
         port.device.baudRate = 115200;
         let stream = port.device.inputStream;
-        let reader = new Windows.Storage.Streams.DataReader(stream);
+        let reader = new streams.DataReader(stream);
         let serialBuffers: pxt.Map<string> = {};
         let readMore = () => {
             // Make sure the device is still active
             if (!activePorts[id]) {
                 return;
             }
-            reader.loadAsync(32).done((bytesRead) => {
-                let msg = reader.readString(Math.floor(bytesRead / 4) * 4);
+            port.readingOperation = reader.loadAsync(32);
+            port.readingOperation.done((bytesRead) => {
+                port.readingOperation = undefined;
+                if (port.cancellingDeferred) {
+                    port.cancellingDeferred.resolve();
+                }
+
+                let msg = reader.readString(Math.floor(reader.unconsumedBufferLength / 4) * 4);
                 pxt.Util.bufferSerial(serialBuffers, msg, id);
-                readMore();
+                setTimeout(() => readMore(), 10);
```

Comment:
why not immediately?

Reply:
Reading from the device should be as much a background task as possible, but I can change the 10 to 1